cli/src/main/java/com/devonfw/tools/ide/common/SystemPath.java

```diff
@@ -166,6 +166,7 @@ public Path getPath(String tool) {
    */
   public void setPath(String tool, Path path) {
 
+    this.paths.add(path);
```

Comment:
No this is the PR where I first stumbled upon this bug.

Reply:
The "fix" is still incorrect. It will add a path entry and cause duplicates.
Assuming you run the same method twice:
```
systemPath.setPath("mvn", "/d/projects/foo/software/mvn");
systemPath.setPath("mvn","/d/projects/bar/software/mvn");
```
This will cause the `PATH` variable to contain both `/d/projects/foo/software/mvn` and `/d/projects/bar/software/mvn`.
You think this scenario sounds unrealistic?
This is exactly what will happen if I call `ide` in project `foo` and then switch to project `bar` and call `ide` there.
Do we have a bug ticket for this problem itself?

Reply:
I just saw your comment right now and couldn't address it in time, good that @MustaphaOuchen took care of it 👍